chain/index/README.MD

```diff
@@ -0,0 +1,188 @@
+
+**Note: If you are a Storage Provider or node operator who does not serve RPC requests, you can skip this document as the `ChainIndexer` is already disabled by default**. 
+
+This document is aimed at RPC providers and node operators who serve RPC requests and aims to walk through the configuration changes, migration flow and operations/maintenance work needed to enable, backfill and maintain the `ChainIndexer`. 
+
+## ChainIndexer Config
+
+The `ChainIndexer` must be enabled on an RPC node as it is disabled by default. Here is the mandatory config to use for all RPC providers to enable the `ChainIndexer` and for ensuring the `EthRPC` and `ActorEventsAPI` are enabled:
+
+```toml
+[ChainIndexer]
+# This is set to false by default which disables the ChainIndexer. 
+# Please ensure that you set it to true before starting your node.
+  EnableIndexer = true 
+
+[Fevm]
+# This is set to false by default which disables the ETH RPC API. 
+# Please ensure that you set it to true before starting your node.
+  EnableEthRPC = true
+
+[Events]
+# This is set to false by default which disables the Actor Events API. 
+# Please ensure that you set it to true before starting your node.
+  EnableActorEventsAPI = true
+```
+
+### Garbage Collection
+
+The `ChainIndexer` includes a garbage collection (GC) mechanism to manage the amount of historical data retained. By default, GC is disabled to preserve all indexed data.
+
+#### Configuration
+
+To configure GC, use the `GCRetentionEpochs` parameter in the `ChainIndexer` section of your config.
+
+The ChainIndexer periodically runs GC if `GCRetentionEpochs` is > 0 and removes indexed data for epochs older than `(current_head_height - GCRetentionEpochs)`.
+
+```toml
+[ChainIndexer]
+  GCRetentionEpochs = X  # Replace X with your desired value
+```
+
+- Setting `GCRetentionEpochs` to 0 (**this is the default**) completely disables GC.
+- Any non-zero value enables GC and determines the number of epochs of historical data to retain.
+
+#### Recommendations
+
+1. **Archival Nodes**: **Keep GC disabled** (`GCRetentionEpochs` = 0) to retain all indexed data.
+
+2. **Non-Archival Nodes**:  Set `GCRetentionEpochs` to match the amount of chain state your node retains 
+(*for example:* if your node is configured to retain 2 days of Filecoin chain state with the Splitstore, set `GCRetentionEpochs` to (number of Filecoin epochs in a day *2) = 5760).
+
+---
+
+**Note: The following config options no longer exist in Lotus and have been removed in favor of the `ChainIndexer` config options explained above:**
+
+```toml
+[Fevm]
+EthTxHashMappingLifetimeDays = 0
+
+[Events]
+DisableHistoricFilterAPI = false
+DatabasePath = ""
+```
```

Comment:
Maybe change it the Note to call out that these legacy configs can be removed when upgrading to the new version:

**Note: The following config options no longer exist in Lotus and have been removed in favor of the `ChainIndexer` config options explained above. They can be removed when upgrading to the Lotus vX.XX.X version"

Reply:
Done. Let's fill in the `vX.XX.X` part when we get closer to the release.